src/main/java/edu/kis/powp/jobs2d/events/SelectHistoryListOptionListener.java

```diff
@@ -0,0 +1,31 @@
+package edu.kis.powp.jobs2d.events;
+
+import javax.swing.JList;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+
+import edu.kis.powp.jobs2d.command.history.HistoryEntry;
+import edu.kis.powp.jobs2d.command.manager.DriverCommandManager;
+
+public class SelectHistoryListOptionListener implements ListSelectionListener {
+
+    private DriverCommandManager commandManager;
+    private final JList<HistoryEntry> historyCommandList;
+
+    public SelectHistoryListOptionListener(DriverCommandManager commandManager,
+            JList<HistoryEntry> historyCommandList) {
+        this.commandManager = commandManager;
+        this.historyCommandList = historyCommandList;
+    }
+
+    @Override
+    public void valueChanged(ListSelectionEvent e) {
+        if (e.getValueIsAdjusting() == false) {
+            HistoryEntry selectedValue = historyCommandList.getSelectedValue();
```

Comment:
I think that 'e' holds info about an index of selected command which could be used to obtain a HistoryEntry from listModel. Please check this.

Reply:
"e" does not provides easy access to chosen element.

Reply:
e doesn't, but I think that a ListModel object does. It would be good to decouple from GUI